src/Compilers/Core/Portable/MetadataReader/PEModule.cs

```diff
@@ -1134,6 +1134,46 @@ internal bool HasIsByRefLikeAttribute(EntityHandle token)
             return null;
         }
 
+#nullable enable
+        internal UnmanagedCallersOnlyAttributeData? TryGetUnmanagedCallersOnlyAttribute(
+            EntityHandle token,
+            IAttributeNamedArgumentDecoder attributeArgumentDecoder,
+            Func<string, TypedConstant, bool, (bool IsCallConvs, ImmutableHashSet<INamedTypeSymbolInternal>? CallConvs)> unmanagedCallersOnlyDecoder)
+        {
+            AttributeInfo info = FindTargetAttribute(token, AttributeDescription.UnmanagedCallersOnlyAttribute);
+            if (!info.HasValue || info.SignatureIndex != 0 || !TryGetAttributeReader(info.Handle, out BlobReader sigReader))
+            {
+                return null;
+            }
+
+            var unmanagedConventionTypes = ImmutableHashSet<INamedTypeSymbolInternal>.Empty;
+
+            if (sigReader.RemainingBytes > 0)
+            {
+                try
+                {
+                    var numNamed = sigReader.ReadUInt16();
+                    for (int i = 0; i < numNamed; i++)
+                    {
+                        // typeCode of the value is checked by the decoder itself
+                        var ((name, value), isProperty, /* typeCode */ _) = attributeArgumentDecoder.DecodeCustomAttributeNamedArgumentOrThrow(ref sigReader);
+                        var namedArgumentDecoded = unmanagedCallersOnlyDecoder(name, value, !isProperty);
+                        if (namedArgumentDecoded.IsCallConvs)
+                        {
+                            unmanagedConventionTypes = namedArgumentDecoded.CallConvs;
+                            break;
```

Comment:
>break; [](start = 28, length = 6)

Is bailing out on the first value consistent with what we do for Obsolete attribute properties?

Reply:
See CrackObsoleteProperties

---
In reply to: [493960320](https://github.com/dotnet/roslyn/pull/47950#discussion_r493960320) [](ancestors = 493960320)

Reply:
Yes. The decoder there takes the first non-null value for each property it is interested in.

---
In reply to: [493961349](https://github.com/dotnet/roslyn/pull/47950#discussion_r493961349) [](ancestors = 493961349,493960320)